app/controllers/application_controller.rb

```diff
@@ -3,11 +3,6 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
-  if Rails.env.production?
```

Comment:
Yep: http://nginx.org/en/docs/http/ngx_http_auth_basic_module.html. It seems to make more sense to centralize the config there rather than in the app code.